### PR TITLE
Fixes #25 Add support for stringifying event data with a maxDepth

### LINKED ISSUE #25: Add support for stringifying event data with a maxDepth
I received some HTTP error 413 calls in the console log. After inspection why the event submission failed, I found out it was about the send in over 5MB of stuff while posting as little as 3 events.

In this case it was caused by the angular integration / feature reporting, based on ngRoute.
Path of the biggest chunks of data data are

```
data.@request.current.locals.$template
```

and also 

```
data.@request.next.locals.$template
```

$template should probably not be included in the events, but I'm not sure if there are others. Best idea might be to pick the properties that should be included, instead of keeping an exclusion list?

### PR DESCRIPTION
**Pull request by niemyjski**
We added the most robust stringify and prune implementation out there. At minimum it's 12x all other implementations.

### CODE REVIEW

File: packages/core/src/Utils.ts [hunk 1]
```diff
@@ -169,102 +189,199 @@ export function endsWith(input: string, suffix: string): boolean {
...
+            return true;
+        }
+
+        // Check for buffer;
+        if ("writeBigInt64LE" in value) {
+          return true;
+        }
+
+        break;
+    }
```
Comment (ejsmith) [inline packages/core/src/Utils.ts:240]: Sure seems like we'd be better off having a list of things that we allow vs trying to keep an exhaustive list of unsupported types when new types can be added and probably different runtimes have different built in types as well. There are only so many intrinsic types that we should support.

**Review by niemyjski — COMMENTED** [review]
Looks good

**Review by ejsmith — COMMENTED** [review]
Seems good.

File: packages/core/src/Utils.ts (hunk 1 above)
Comment (niemyjski) [reply packages/core/src/Utils.ts:240]: Yeah, we can work towards improving this. I'm assuming there will be more types over time, and we will need to optimize this some.

**Pull request merged**